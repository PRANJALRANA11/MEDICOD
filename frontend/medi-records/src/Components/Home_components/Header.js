import React from 'react'
import '../../Styles/Home-styles/Header.css'; 
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <>
     <header className='home_compo_header'>
        <div className='header_div'>
            <div className='heading_div'>
                <div style={{
                    backgroundColor:"rgb(15 22 41)",
                    borderRadius:"10px",
                    padding:"4px 12px",
                    width:"360px",
                    fontFamily:"'Poppins', sans-serif",
                            }}>
                    <span>
                        All Your Medical Records At One Place
                    </span>
                </div>
                <div style={{
                        width:"38rem",
                            }}>
                    <h1 style={{
                        fontSize:"3rem",
                        fontFamily:"'Poppins",
                        color:"#e1e7ef",
                                }}>
                        Effortless Medical Record Management Online
                    </h1>
                </div>
                <div
                style={{
                    width:"47rem",
                       }}>
                    <span
                    style={{
                        color:"#7f8ea3",
                        fontSize:"1.2rem",
                        marginLeft:"-15px",
                            }}>
                        Welcome to MEDICOD, 
                        where your health information is our top priority.
                        We offer a secure and user-friendly platform to store 
                        and manage your medical records online.
                    </span>
                </div>
                <div
                style={{
                    backgroundColor:"#fff",
                    borderRadius:"10px",
                    width:"109px",  
                    padding:"10px 20px",
                    fontWeight:"500",
                    fontFamily:"'Poppins', sans-serif",
                    marginTop:"30px",
                    marginLeft: "17px",
                        }}>
                    <Link to='/Dashboard' style={{
                        textDecoration:"none",
                        color:"#000",
                                        }}>
                        Get Started
                    </Link>
                </div>
                <div style={{
                    display:"flex",
                    marginTop:"50px",
                    marginLeft: "17px",
                            }}>
                    <div style={{
                        display:"flex",
                        marginRight:"30px",
                                }}>
                        <div style={{
                            backgroundColor:"#fff",
                            width:"30px",
                            height:"30px",
                            padding:"6px",
                            borderRadius:"5px",
                                    }}>
                            <i className="fa-regular fa-user fa-xl" ></i>
                        </div>
                            <span style={{
                                fontSize:"1.3rem",
                                marginLeft:"10px",
                                color:"#e1e7ef",
                                fontFamily:"'Poppins', sans-serif",
                                        }}>
                                Over 1000+ clients
                            </span>
                    </div>
                    <div style={{
                        display:"flex",
                        marginRight:"30px",
                                }}>
                        <div style={{
                            backgroundColor:"#fff",
                            width:"30px",
                            height:"30px",
                            padding:"6px",
                            borderRadius:"5px",
                                    }}>
                            <i className="fa-regular fa-circle-check fa-xl"></i>
                        </div>    
                            <span style={{
                                fontSize:"1.3rem",
                                marginLeft:"10px",
                                color:"#e1e7ef",
                                fontFamily:"'Poppins', sans-serif",
                                        }}>
                                Verified Trusted
                            </span>
                    </div>
                    <div style={{
                        display:"flex",
                                }}>
                        <div style={{
                            backgroundColor:"#fff",
                            width:"30px",
                            height:"30px",
                            padding:"6px",
                            borderRadius:"5px",
                                    }}>
                            <i className="fa-solid fa-shield fa-xl"></i>
                        </div>    
                            <span style={{
                                fontSize:"1.3rem",
                                marginLeft:"10px",
                                color:"#e1e7ef",
                                fontFamily:"'Poppins', sans-serif",
                                        }}>
                                Secure & Safe
                            </span>
                    </div>
                </div>
            </div>
            <div className='img_div'>
                <img src="/assets/doctor.png" alt='doctor_image'
                style={{
                    width:"60%",
                    height:"auto",
                    marginLeft:"200px",
                    marginTop:"70px",
                        }}>
                </img>
            </div>
        </div>
     </header> 
    </>
  )
}
