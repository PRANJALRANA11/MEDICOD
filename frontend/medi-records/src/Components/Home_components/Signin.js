import React, { useState } from 'react';
import { signin_api } from '../../api/api';
import '../../Styles/Home-styles/Authentication.css';
import { Link, useNavigate } from 'react-router-dom';

export default function Signin() {
  let navigate = useNavigate();

  const [formData, setFormData] = useState({
    email: '',
    password: '',
  });

  const [errors, setErrors] = useState({});

  const handleFormData = (e) => {
    const { name, value } = e.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const validateForm = () => {
    let valid = true;
    const newErrors = {};

    if (formData.email.trim() === '') {
      newErrors.email = 'Email is required';
      valid = false;
    }

    if (formData.password.trim() === '') {
      newErrors.password = 'Password is required';
      valid = false;
    }

    setErrors(newErrors);
    return valid;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();

    if (validateForm()) {
      try {
        const response = await signin_api(formData);
        if (response === 200) {
          console.log(response.data);
          navigate('/Dashboard');
        }
      } catch (error) {
        setErrors({ email: 'Invalid Credentials' });
        console.error('Error during login:', error);
      }
    }
  };

  return (
    <div id="Body_signup">
      <div className="signup">
        <h1 className="signup_heading">Welcome To Medicod</h1>
        <form className="form" onSubmit={handleSubmit} method="POST">

        {errors.email && (
            <div className="error" style={{ color: 'red', paddingBottom: '1rem' }}>
              {errors.email}
            </div>
          )}

          <input
            placeholder="Email"
            name="email"
            type="email"
            required={true}
            value={formData.email}
            onChange={handleFormData}
          />
       

          <input
            placeholder="Password"
            name="password"
            type="password"
            required={true}
            value={formData.password}
            onChange={handleFormData}
          />
          {errors.password && (
            <div className="error" style={{ color: 'red', paddingBottom: '1rem' }}>
              {errors.password}
            </div>
          )}

          <Link id="Forget-pass" to="/forgotpassword">
            Forgot Password ?
          </Link>
          <button id="Login_btn" type="submit">
            Login
          </button>
          <div>
            <span>
               not signed up yet ? <Link to="/Signup">Sign up now</Link>
            </span>
          </div>
        </form>
      </div>
    </div>
  );
}
