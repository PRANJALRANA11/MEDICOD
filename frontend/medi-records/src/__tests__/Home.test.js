// app.test.js
import  React from 'react'
import {render, screen,fireEvent} from '@testing-library/react'
import Navbar from '../Components/Home_components/Navbar'
import Signin from '../Components/Home_components/Signin'
import Signup from '../Components/Home_components/Signup'
import Header from '../Components/Home_components/Header'
import { MemoryRouter } from 'react-router-dom'
describe('Navbar Component', () => {
  it('renders without crashing', () => {
    render(<MemoryRouter><Navbar /></MemoryRouter>);
    expect(screen.getByText('MEDICOD')).toBeInTheDocument();
  });

  it('displays the "Signup" link when not authenticated', () => {
    render(<MemoryRouter><Navbar /></MemoryRouter>);
    expect(screen.getByText('Signup')).toBeInTheDocument();
  });
  it('matches the snapshot', () => {
    const { asFragment } = render(<MemoryRouter><Navbar /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Signin Component', () => {
  it('renders the form fields and submit button', () => {
    render(<MemoryRouter><Signin /></MemoryRouter>);
    expect(screen.getByPlaceholderText(/Email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Password/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Login/i })).toBeInTheDocument();
  });

  it('calls the submit function with the form data when submitted', async () => {
     render(<MemoryRouter><Signin /></MemoryRouter>);
    fireEvent.change(screen.getByPlaceholderText(/Email/i), { target: { value: 'test@example.com' } });
    fireEvent.change(screen.getByPlaceholderText(/Password/i), { target: { value: 'password123' } });
    fireEvent.click(screen.getByRole('button', { name: /Login/i }))
  });
  it('matches the snapshot', () => {
    const { asFragment } = render(<MemoryRouter><Signin /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Signup Component', () => {
  it('renders the form fields and submit button', () => {
    render(<MemoryRouter><Signup /></MemoryRouter>);
    expect(screen.getByPlaceholderText(/Enter Your Name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Enter Your Email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Create Password/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Sign Up/i })).toBeInTheDocument();
  });

  it('calls the submit function with the form data when submitted', async () => {
    render(<MemoryRouter><Signup /></MemoryRouter>);
      fireEvent.change(screen.getByPlaceholderText(/Enter Your Name/i), { target: { value: 'John Doe' } });
      fireEvent.change(screen.getByPlaceholderText(/Enter Your Email/i), { target: { value: 'test@example.com' }});
      fireEvent.change(screen.getByPlaceholderText(/Create Password/i), { target: { value: 'password123' }});
      fireEvent.click(screen.getByRole('button', { name: /Sign Up/i }));
    });
    it('matches the snapshot', () => {
      const { asFragment } = render(<MemoryRouter><Signup /></MemoryRouter>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

describe('Header Component', () => {
  it('renders the header component', () => {
    render(<MemoryRouter><Header /></MemoryRouter>);

    // Check if the header text is displayed
    expect(screen.getByText('All Your Medical Records At One Place')).toBeInTheDocument();
    expect(screen.getByText('Effortless Medical Record Management Online')).toBeInTheDocument();
    // Check if the "Get Started" link is present
    const getStartedLink = screen.getByText('Get Started');
    expect(getStartedLink).toBeInTheDocument();
    expect(getStartedLink).toHaveAttribute('href', '/Dashboard');

    // Check if the client information icons and text are displayed
    expect(screen.getByText('Over 1000+ clients')).toBeInTheDocument();
    expect(screen.getByText('Verified Trusted')).toBeInTheDocument();
    expect(screen.getByText('Secure & Safe')).toBeInTheDocument();

    // Check if the doctor image is displayed
    const doctorImage = screen.getByAltText('doctor_image');
    expect(doctorImage).toBeInTheDocument();
  });
});
