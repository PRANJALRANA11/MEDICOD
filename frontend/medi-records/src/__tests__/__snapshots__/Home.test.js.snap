// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <a
          href="/"
        >
          MEDICOD
        </a>
      </div>
      <ul
        class="navbar-links"
      >
        <li>
          <a
            href="#About"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#Contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            href="/signup"
          >
            Dashboard
          </a>
        </li>
        <li>
          <a
            href="/signup"
            style="padding: 8px 16px; border: 1px solid #1d283a; border-radius: 10px;"
          >
            Signup
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Signin Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="Body_signup"
  >
    <div
      class="signup"
    >
      <h1
        class="signup_heading"
      >
        Welcome To Medicod
      </h1>
      <form
        class="form"
        method="POST"
      >
        <input
          name="email"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <input
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <a
          href="/forgotpassword"
          id="Forget-pass"
        >
          Forgot Password ?
        </a>
        <button
          id="Login_btn"
          type="submit"
        >
          Login
        </button>
        <div>
          <span>
            not signed up yet
            <a
              href="/Signup"
            >
              Sign up now
            </a>
          </span>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Signup Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="Body_signup"
  >
    <div
      class="signup"
    >
      <h1
        class="signup_heading"
      >
        Welcome To Medicod
      </h1>
      <form
        class="form"
        method="POST"
      >
        <input
          name="name"
          placeholder="Enter Your Name"
          required=""
          type="text"
          value=""
        />
        <input
          name="email"
          placeholder="Enter Your Email"
          required=""
          type="email"
          value=""
        />
        <input
          name="password"
          placeholder="Create Password"
          required=""
          type="password"
          value=""
        />
        <button
          id="Login_btn"
          type="submit"
        >
          Sign Up
        </button>
        <div>
          <span>
            Already have an account? 
            <a
              href="/Signin"
            >
              Sign in now
            </a>
          </span>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
